logging.go

```diff
@@ -699,7 +700,13 @@ type defaultCustomLog struct {
 // and enables INFO-level logs and higher.
 func newDefaultProductionLog() (*defaultCustomLog, error) {
 	cl := new(CustomLog)
-	cl.writerOpener = StderrWriter{}
+	f := flag.Lookup("test.v")
+	if (f != nil && f.Value.String() != "true") || strings.Contains(os.Args[0], ".test") {
+		cl.writerOpener = &DiscardWriter{}
+	} else {
+		cl.writerOpener = StderrWriter{}
+	}
```

Comment:
I think we do want stderr logs during tests tho, cause it can help trace why a failure happened. Pretty sure `go test` skips outputting them if there's no failure (unless `-v` is set)

Reply:
sounds fine to me. should be resolved.

i will say that i do find it somewhat odd that caddy logs in tests by default. i generally expect `go test` to run silently.